components/scream/cime_config/buildnml

```diff
@@ -47,52 +48,144 @@ def do_cime_vars(entry, case):
+                all_match = True
+                for k, v in selectors.items():
+                    val_re = re.compile(v)
+
+                    if k in ez_selectors:
+                        ez_env, ez_regex = ez_selectors[k]
+                        case_val = case.get_value(ez_env)
+                        expect(case_val is not None, "Could not get selector value {} from case".format(ez_env))
+
+                        ez_regex_re = re.compile(ez_regex)
+                        m = ez_regex_re.match(case_val)
+                        if m:
+                            val = m.groups()[0]
+                        else:
+                            val = "DOES NOT MATCH"
+
+                    else:
+                        val = case.get_value(k)
+                        expect(val is not None, "Could not get selector value {} from case".format(k))
+
+
+                    if val_re.match(val) is None:
+                        all_match = False
+                        break
+
+                if all_match:
+                    orig_child = child_values[child_name]
+                    orig_child.text = do_cime_vars(child_val, case)
+
+                children_to_remove.append(child)
+
             else:
-                segments[-1] += entry[idx]
-
-        expect(idx < len(entry), "Switch statement parse error in string '{}'".format(entry))
-        end_switch_idx = idx + 1
-        best_val = None
-        for segment in segments:
-            components = segment.split("=>", 1)
-            if (len(components) == 1 or '<' in components[0]):
-                if best_val is None:
-                    best_val = do_switches(segment).strip()
+                expect(child_name not in child_values, "Repeat of child {} without selectors".format(child_name))
+                child_values[child_name] = child
+                child.text = do_cime_vars(child_val, case)
+
+        else:
+            evaluate_selectors(child, case, ez_selectors)
+
+    for child_to_remove in children_to_remove:
+        element.remove(child_to_remove)
+
+###############################################################################
+def create_raw_xml_file(case, caseroot):
+###############################################################################
+    src = os.path.join(case.get_value("SRCROOT"), "components/scream/cime_config/namelist_defaults_scream.xml")
+
+    raw_xml_file = os.path.join(caseroot, "namelist_scream.xml")
+    if os.path.exists(raw_xml_file):
+        print("{} already exists, will not overwrite. Remove to regenerate".format(raw_xml_file))
+    else:
+        with open(src, "r") as fd:
+            tree = ET.parse(fd)
+            root = tree.getroot()
+
+        selectors = None
+        selectors_elem = None
+        for child in root:
+            if selectors is None:
+                expect(child.tag == "selectors", "Unexpected child.tag {}".format(child.tag))
```

Comment:
What this means is that we expect the first big organizational block of XML in `namelist_defaults_scream.xml` to be the block that defines our "easy selectors". This block is currently mandatory as I think we agreed that we will want "easy selectors" like those available in eam's nml config files. If we did decide to go with a definitions.xml file, this would be moved there.

Reply:
Ah, I see now. The `if selectors is None:` line forces this check to run only on the first entry of the XML. Got it.